### DISCUSSION TITLE
Issues daisy chaining Shift Register Out

### DISCUSSION THREAD

**Discussion by dav09012** [tttapa/Control-Surface#737] [Q&A] [answered]
I'm trying to control a series of LEDs via a program called GrandOrgue. I want specific LEDs to light up when certain note values. I have been successful with using 1 shift register to control 8 LEDs, but I can't seem to get it to work when I try to daisy chain multiple Shift Registers together. 
I'm using the SN74HC595N, and here is my code below. I've connected pin #9 from Shift Register 1 to Pin #14 of the second Shift Register. Pins 8 and 13 go to ground. Pins 10 and 16 are connected to vcc. Pin 12 goes to the arduino pin 10, and 11 goes to arduino #52. 
The issue I'm having is that the first 8 LEDs behave normally. They turn on and off with the notes I've mapped them to. The rest of the LEDs remain lit all the time and I have no control over them. 
Any help would be appreciated! 
Thanks, 
```cpp
// Include the library
#include <Control_Surface.h>

// Instantiate a MIDI Interface to use
USBMIDI_Interface midi;

// Instantiate a shift register as output for the LEDs
SPIShiftRegisterOut<8*3> sreg = {
  10,       // Latch pin (ST_CP)
  MSBFIRST, // Byte order
};

using namespace MIDI_Notes;

// Create an array of LEDs that listen to MIDI Note messages, turning on and off
// the LEDs connected to the eight output pins of the shift register
NoteValueLED leds[] = {
  {sreg.pin(0), {36, CHANNEL_5, } }, // LED pin, address (note number, channel, cable)
  {sreg.pin(1), {37, CHANNEL_5, } }, //
  {sreg.pin(2), {38, CHANNEL_5, } }, //
  {sreg.pin(3), {39, CHANNEL_5, } }, //
  {sreg.pin(4), {40, CHANNEL_5, } }, //
  {sreg.pin(5), {41, CHANNEL_5, } }, //
  {sreg.pin(6), {42, CHANNEL_5, } }, //
  {sreg.pin(7), {43, CHANNEL_5, } }, //
  {sreg.pin(8), {44, CHANNEL_5, } }, // 
  {sreg.pin(9), {45, CHANNEL_5, } }, //
  {sreg.pin(10), {46, CHANNEL_5, } }, //
  {sreg.pin(11), {47, CHANNEL_5, } }, //
  {sreg.pin(12), {48, CHANNEL_5, } }, //
  {sreg.pin(13), {49, CHANNEL_5, } }, //
  {sreg.pin(14), {50, CHANNEL_5, } }, //
  {sreg.pin(15), {51, CHANNEL_5, } }, //
  {sreg.pin(16), {52, CHANNEL_5, } }, //
  {sreg.pin(17), {53, CHANNEL_5, } }, //
  {sreg.pin(18), {54, CHANNEL_5, } }, //
  {sreg.pin(19), {55, CHANNEL_5, } }, //
  {sreg.pin(20), {56, CHANNEL_5, } }, //
  {sreg.pin(21), {57, CHANNEL_5, } }, //
  {sreg.pin(22), {58, CHANNEL_5, } }, //
  {sreg.pin(23), {59, CHANNEL_5, } }, //
};



// Initialize the Control Surface
void setup() {
  Control_Surface.begin();
}

// Update the Control Surface
void loop() {
  Control_Surface.loop();
}
```

**Comment by Veroledez** [comment, marked as answer]
Hola, ya probaste los shift registers por separado, puede que el segundo este dañado, o la salida del primero este dañada y se pierde la comunicaciob  poreso los otros no responden, intenta probar cada uno y descartar algun defectuoso

**Comment by tttapa** [comment]
I'm unable to reproduce the issue you describe using the latest `master` version. If you upgrade, you'll have to specify the SPI interface:
```cpp
SPIShiftRegisterOut<8*3> sreg = {
  SPI,
  10,       // Latch pin (ST_CP)
  MSBFIRST, // Byte order
};
```

If that still doesn't work, I suspect a hardware issue. Try the [2.RGB-LED-Chaser.ino](https://tttapa.github.io/Control-Surface-doc/Doxygen/df/d9e/2_8RGB-LED-Chaser_8ino-example.html) example and the wiring therein.

**Comment by dav09012** [comment]
Thanks for the responses. Veroledez was right, it was an issue with the shift register. I don't know why I didn't try swapping it earlier. I plugged a new one in and it worked perfectly.
